server/base/src/main/java/org/apache/accumulo/server/util/TabletIterator.java

```diff
@@ -101,6 +102,20 @@ public boolean hasNext() {
 
       currentTabletKeys = scanToPrevEndRow();
       if (currentTabletKeys.size() == 0) {
+        if (lastTablet != null) {
```

Comment:
A brief human-friendly comment might be helpful to explain what this block is trying to check for:

```suggestion
        // Always expect the default tablet to exist for a table. The following checks for the case when
        // the default tablet was not seen when it should have been seen.
        if (lastTablet != null) {
```

Comment copied from #1266